cpp/deplex/src/deplex/plane_extractor.cpp

```diff
@@ -303,20 +316,17 @@ std::vector<CellSegment> PlaneExtractor::Impl::createPlaneSegments(CellGrid cons
     }
     // 3. Grow seed
     CellSegment plane_candidate(cell_grid[seed_id]);
-    std::vector<bool> activation_map = growSeed(seed_id, unassigned_mask, cell_grid);
+    std::vector<size_t> cells_to_merge = growSeed(seed_id, unassigned_mask, cell_grid);
 
     // 4. Merge activated cells & remove from hist
-    for (size_t i = 0; i < activation_map.size(); ++i) {
-      if (activation_map[i]) {
-        plane_candidate += cell_grid[i];
-        hist.removePoint(static_cast<int32_t>(i));
-        unassigned_mask[i] = false;
-        --remaining_planar_cells;
-      }
+    for (auto& v : cells_to_merge) {
```

Comment:
Using reference for primitive type (e.g. int) is more expensive than using its copy. It's better to remove `&`